presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java

```diff
@@ -348,11 +348,13 @@ public static Path getTableDefaultLocation(String user, HiveMetastore metastore,
         }
 
         Path databasePath = new Path(location);
-        if (!pathExists(user, hdfsEnvironment, databasePath)) {
-            throw new PrestoException(HIVE_DATABASE_LOCATION_ERROR, format("Database '%s' location does not exist: %s", schemaName, databasePath));
-        }
-        if (!isDirectory(user, hdfsEnvironment, databasePath)) {
-            throw new PrestoException(HIVE_DATABASE_LOCATION_ERROR, format("Database '%s' location is not a directory: %s", schemaName, databasePath));
+        if (!location.startsWith("s3")) {
```

Comment:
Comparing the URI is brittle and will break if a new URI scheme is registered for S3. There is a method `HiveLocationService.shouldUseTemporaryDirectory()` that already does the check correctly.

Let's create a new helper method in `HiveUtil` named `isS3FileSystem` using the code from `shouldUseTemporaryDirectory`, then have `shouldUseTemporaryDirectory` call the new method.

Then we can use the new method here as well.